src/constraint_system/standard/composer.rs

```diff
@@ -587,31 +613,50 @@ impl StandardComposer {
         self.q_m.push(q_m);
         self.q_o.push(q_o);
         self.q_c.push(q_c);
-        self.q_4.push(Scalar::zero());
+        self.q_4.push(q_4);
 
         self.public_inputs.push(pi);
 
-        self.perm
-            .add_variables_to_map(a, b, c, self.zero_var, self.n);
+        self.perm.add_variables_to_map(a, b, c, d, self.n);
 
         self.n = self.n + 1;
 
         c
     }
+
     // q_m * a * b - c = 0
     pub fn mul(&mut self, q_m: Scalar, a: Variable, b: Variable, pi: Scalar) -> Variable {
+        self.big_mul(q_m, a, b, (Scalar::zero(), self.zero_var), pi)
+    }
+
+    // q_m * a * b + (q_4 * d) - c = 0
```

Comment:
You can't multiply the four inputs together? Thought this was the goal of the big mul.

Reply:
Mentioned in the previous PR; this will affect the degree of the quotient polynomial when the other custom gates are added. There may be a way to do it, if the equation is modified or the blow up is allowed

Reply:
True, sorry for repeating it then. Was just info purposes.